core/src/main/kotlin/com/mineinabyss/geary/ecs/query/Query.kt

```diff
@@ -100,8 +105,14 @@ public abstract class Query : Iterable<QueryResult> {
     }
 
     protected inline fun <reified T : GearyComponent> has(set: Boolean = true): GearyEntity {
-        val componentId = componentId<T>().let { if (set) it and HOLDS_DATA.inv() else it }
+        val componentId = componentId<T>().let { if (set) it or HOLDS_DATA else it }
```

Comment:
As I understand:
We need "or" because _it_ is non-data component id, so we have to change it to one

Reply:
Ah yep, surprised i haven't caugh't that earlier in any tests...

I'm currently rewriting this entire class and adding some new features to the query selection. One of those is simplifying `has` to not care for set/not set data, hopefully I can get that in soon!